### DISCUSSION TITLE
How to know the implemented rewrite rules in cvc5?

### DISCUSSION THREAD

**Discussion by merlinsun** [cvc5/cvc5#10275] [Q&A] [answered]
Hello!
Is there a dedicated function in cvc5 that allows for the display of all implemented rewrite rules? While I understand that examining the source code is an option, it can be somewhat time-consuming and requires extra effort.

**Comment by martin-cs** [comment, marked as answer]
There isn't really a function for this because some of the rewrite rules aren't easily described like that.  You might want to look at some of the work that has been done on proof generation as they have formalisations of some of the rewrite rules.  What are you looking to with the rewrite rules?
